src/summarycode/score.py

```diff
@@ -304,38 +401,57 @@ def check_declared_licenses(license_match_objects):
     """
     return any(is_good_license(license_match_object) for license_match_object in license_match_objects)
 
-
-def get_field_values_from_codebase_resources(
-    codebase,
+def get_field_values_from_resources(
+    resources,
     field_name,
     key_files_only=False,
-    is_string=False
+    is_string=False,
+    is_codebase=False
 ):
     """
     Return a list of values from the `field_name` field of the Resources from
-    `codebase`
-
+    the provided `resources`.
+    
     If `key_files_only` is True, then we only return the field values from
     Resources classified as key files.
 
     If `key_files_only` is False, then we return the field values from Resources
     that are not classified as key files.
     """
     values = []
-    for resource in codebase.walk(topdown=True):
-        if key_files_only:
-            if not resource.is_key_file:
-                continue
-        else:
-            if resource.is_key_file:
-                continue
-        if is_string:
-            value = getattr(resource, field_name, None) or None
-            if value:
-                values.append(value)
-        else:
-            for value in getattr(resource, field_name, []) or []:
-                values.append(value)
+    
+    if is_codebase:
+        for resource in resources.walk(topdown=True):
```

Comment:
```suggestion
        for resource in codebase.walk(topdown=True):
```

This is wrong and extremely confusing code. You cannot store a codebase object in a `resources` variable and walk/get attributes from it depending on a flag value. You should only pass `resources` here from before, and these should always be resource objects consistently. See other comments on this.

Reply:
Resolved